Phoenix/Presence.cs

```diff
@@ -39,7 +45,296 @@ namespace Phoenix {
 	//     }
 	//     let onlineUsers = Presence.list(state, listBy)
 	//
+
 	public class Presence {
-		// TODO
+
+
+		public static JObject syncState(JObject currentState, JObject newState, Action<string, JToken, JToken> onJoin = null, Action<string, JToken, JToken> onLeave = null){
+			JObject state = (JObject)JObject.Parse(currentState.ToString());
```

Comment:
This is serializing currentState to string, then back to JObject. Perhaps there is a better way to clone the object if needed.